src/04-modality-specific-files/05-task-events.md

```diff
@@ -93,29 +114,76 @@ Example:
+when applicable:
+
+| Field name           | Description                                                                                                                                                                                                                                                        |
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| StimulusPresentation | RECOMMENDED. [Object][] containing key value pairs related to the software used to present the stimuli during the experiment, specifically: `OperatingSystem`, `SoftwareName`, `SoftwareRRID`, `SoftwareVersion` and `Code`. See table below for more information. |
+
+The object supplied for `StimulusPresentation` SHOULD include the following key-value pairs:
+
+| Key name        | Description                                                                                                                                                                                                                   |
+|-----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| OperatingSystem | RECOMMENDED. Operating system used to run the stimuli presentation software (for formatting recommendations, see examples below this table).                                                                                  |
+| SoftwareName    | RECOMMENDED. Name of the software that was used to present the stimuli.                                                                                                                                                       |
+| SoftwareRRID    | RECOMMENDED. [Research Resource Identifier](https://scicrunch.org/resources) of the software that was used to present the stimuli. Examples: The RRID for Psychtoolbox is 'SCR_002881', and that of PsychoPy is 'SCR_006571'. |
+| SoftwareVersion | RECOMMENDED. Version of the software that was used to present the stimuli.                                                                                                                                                    |
+| Code            | RECOMMENDED. [URI][uri] of the code used to present the stimuli. Persistent identifiers such as DOIs are preferred. If multiple versions of code may be hosted at the same location, revision-specific URIs are recommended.  |
+
+The operating system description should include:
+-   type (Windows, MacOS, Linux),
+-   distribution (if applicable: Ubuntu, Debian)
+-   the version number or equivalent code name
+
+Examples:
+-   Windows - 10
+-   macOS - Catalina or macOS - 10.15
```

Comment:
That's not specific enough, it should include the minor version too, i.e.,

```suggestion
-   Windows - 10 - 20H2
-   macOS - 10.15.6
```

Reply:
I agree that that'd be good.

Do you have a link to where "windows" versions are described? I had never heard about  this "20H2" convention before (or seen it).

Reply:
Sorry, 20H2 is just an insider preview build. The public releases carry much simpler names.
https://en.wikipedia.org/wiki/Windows_10_version_history

Reply:
Ah thanks, then we should probably say something like `Windows - 10 - 2004`

Reply:
Is there any specific reason why we're using the hyphens here? I think they're unnecessary… and from a typographical POV, these should be – anyway ;)

Reply:
no particular reason we were thinking about how we can specify the OS sufficiently for reproducing an experiment, while at the same time not over-engineering the whole thing.

Do you have a new / better suggestion? :)

Reply:
maybe a JSON object with  `type`, `distribution` (optional), `version`

... then maybe Windows 10 could be of type windows, with distribution 10, and version 2004

similar for osx: type macos, no distribution, version 10.15.6

or linux: type linux, ubuntu, 18.04.5


`type` and `version` are REQUIRED, `distribution` is optional?

WDYT?

Reply:
I guess I would just remove the hyphens and stress that ideally, the full version number should be specified. For example:

```
- Windows 10
- Windows 10 1903 (better)
- macOS 10.15
- macOS 10.15.6  (better)
```
WDYT?

Reply:
also, what about 32bit and 64 bit :bomb: :exploding_head:

Reply:
And Apple is soon going to start rolling out ARM chips. So yes, maybe we should allow for adding architecture.

(re 32/64 bits specifically, who on earth would still run a 32-bit OS? oh, right, scientists :S)

Reply:
We are now at exactly the point that we wanted to avoid in the first place ... I pushed a commit to settle the issue, I think the specification is now fair enough. 

For somebody to really re-create the experiment as it was run, much much more information would be needed anyways (display kind, size, ... chinrest yes or no ... distance of eyes from screen, ...). And I think that is out of the scope of BIDS, which is about sharing data and not making everything reproducible from an idea over funding acquisition to experimentation, analysis, etc...

having that said, I think it **is** good, to include pointers in BIDS that help to understand the data shipped with BIDS. So we **do** have some fields about funding, ethics, ... and now also about the experiment code used to generate the data --> which I find helpful and good and in the scope of BIDS.

What are your opinions @hoechenberger @Remi-Gau

Reply:
Good summary, @sappelhoff! I wholeheartedly agree. Let's merge this and move on :)

Reply:
YES !! 

BIDS metadata is not meant to be a substitute for a method section.

This PR was inspired by some of the COBIDAS recommendations and I am sure that they don't mention chips architecture, so let's not go down that slippery slope.